Automattic/woocommerce-payments
# Sanitize option key for subscription

Replace spaces with underscores between words in option key names for sub products.

Fixes #4259 

#### Changes proposed in this Pull Request
Sanitize Subscription product option key name before they are saved in DB

Option keys, for example, tax names, are saved as `_wcpay_product_id_Tax name`  instead of `_wcpay_product_id_tax_name`. The PR replaces spaces with underscores, and makes the option name in all small case, before it gets concatenated and saved in the database. 


#### Testing instructions

* Turn taxes on for a WC store
* Add a new tax with a name containing spaces. For example ('Tax for climate')
* Apply the tax to a WC Pay subscriptions product (or leave it on all store products)
* Purchase the product with the tax applied to it
* Verify the wp_options table. There should be a new option called `_wcpay_product_id_test_tax_for_climate` (contains name of the tax with underscore instead of spaces)

-------------------



Post merge
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable. - (Not applicable)
